Example-CustomUserGroups-REXX/USRGRP01.rexx

```diff
@@ -0,0 +1,2980 @@
+      targetUserGroupURI,,
+      lparURI,,
+      CPCtargetName,
+    )
+  End
+  /********************************************************************
+   * Add the target LPAR to the Custom User Group
+   ********************************************************************/
+  Else If addLPARToCustomUserGroup == TRUE Then Do
+    Say 'Adding new member to the Custom User Group'
+    addedNewGroupMember = AddNewGroupMember(,
+      targetUserGroupURI,,
+      lparURI,,
+      CPCtargetName,
+    )
+  End
+
+  If (,
+    removeLPARFromCustomUserGroup == TRUE |,
+    addLPARToCustomUserGroup == TRUE,
+  ) Then Do
+    /*
+     * Verify the action was taken to either add or remove the LPAR
+     * from the target user group
+     */
+    didRetrieveMembers = GetCustomUserGroupMembers(,
+      CPCtargetName,,
+      targetUserGroupURI,
+    )
+  End
+
+  Call Cleanup
+
+Return 0
+
+/**********************************************************************
+ * Function: GetArgs
+ *
+ * Purpose: Parse user's arguments and set the appropriate global
+ *          variables
+ *
+ * Input: ARG_STR - The string representation of the arguments passed
+ *                  to the REXX script by the user
+ *
+ * Output: An Integer indicating if function was successful
+ *         (0 if successful, -1 if not.)
+ *
+ * Side-Effects:
+ *
+ *  1. Calls the Usage helper function to display the script's expected
+ *     Usage. This Usage helper function returns a -1
+ *  2. Successful parsing of the user's arguments will set the
+ *     appropriate global variables
+ *
+ **********************************************************************/
+GetArgs:
+
+  MAX_NUM_OF_ARGS = 10
+
+  ARG_STR = Arg(1)
+
+  argCount = Words(ARG_STR)
+
+  If argCount > MAX_NUM_OF_ARGS Then
+    Return Usage( 'Wrong number of arguments' )
+
+  Do i = 1 to argCount
+    userArg = Word(ARG_STR, i)
+
+    If Translate(userArg) == '-C' Then Do
+      i = i + 1
+
+      If i > argCount Then Do
+        Return Usage('-C option specified, but is missing CPC name')
+      End
+
+      CPCname = Word(ARG_STR, i)
+      localCPC = False
+
+    End
+    Else If Translate(userArg) == '-L' Then Do
+      i = i + 1
+
+      If i > argCount Then Do
+        Return Usage('-L option specified, but is missing LPAR name')
+      End
+
+      LPARname = Word(ARG_STR, i)
+      localLPAR = False
+
+    End
+    Else If Translate(userArg) == '-G' Then Do
+      i = i + 1
+
+      If i > argCount Then Do
+        Return Usage(,
+          '-G option specified, but is missing Custom User Group',
+        )
+      End
+
+      targetUserGroup = Word(ARG_STR, i)
+      isMissingTargetGroup = FALSE
+
+    End
+    Else If Translate(userArg) == '-A' Then Do
+      i = i + 1
+
+      If i > argCount Then Do
+        Return Usage(,
+          '-A option specified, but is missing an Action to take',
+        )
+      End
+
+      /*
+       * Convert the fourth argument (ADD or REMOVE) to uppercase for

```

Comment:
Not necessary but from a design standpoint I would recommend that you add a third action of 'LIST' as the default action when action is not specified also allowing the user to specify the this action explicitly.